addons/account_edi/models/account_edi_format.py

```diff
@@ -488,3 +488,10 @@ def _retrieve_currency(self, code):
     def _format_error_message(self, error_title, errors):
         bullet_list_msg = ''.join('<li>%s</li>' % msg for msg in errors)
         return '%s<ul>%s</ul>' % (error_title, bullet_list_msg)
+
+    def _get_edi_attachments(self, doc, record_data):
```

Comment:
Do this in mail_template directly. account_edi_format should not have to handle documents, and even less record_data from mail.

Reply:
You are right

Reply:
iI think this method should be in the format object but receive the recordset account.edi.document itself don't you think?

This was an error from @umiphos the actual feature was asked like that.

Reply:
This is debatable, but I would like to avoid that we put anything in account_edi_format. It should be there only if there is an interest to override it for other formats, which might be the case in the future, I agree, but I would like to keep things simple for now and not overburden account_edi_format. Moreover, account_edi_format should only be concerned with the invoices (or account.move), and the web-service if there's one, but email related stuff should be put in the related model, that is mail_template (which doesn't prevent you from doing specific stuff based on the format).
More broadly, we would like to avoid edi_formats to manage edi_documents directly because it creates a circular reference. edi_documents do stuff with edi_formats, not the other way around.

Reply:
Good point, mail.template is then!

Reply:
@bfr-o @jco-odoo ready this and for merge.

Reply:
@umiphos You'll also need to squash your commits into one with a nice commit message.

cc @nhomar